### PR TITLE
Comprehensive Error Handling, Logging, and Workflow Improvements

### PR DESCRIPTION
**Pull request by db-tycoon-stephen**
# Streamline Logging and Error Handling

## Overview
Improved logging, error handling, and configuration across the OSAA MVP pipeline.

### Pre-commit Configuration
- Reduced hooks to essential checks
- Simplified validation process

### Logging Improvements
- Centralized error logging function
- Reduced verbosity of error messages
- Simplified troubleshooting steps

### AWS Credentials Handling
- Improved error message clarity
- Simplified error handling
- Maintained critical error reporting

## Key Improvements
- Less complex logging
- Clearer error messages
- More efficient configuration management

### Pre-commit Config
```yaml
repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v4.4.0
    hooks:
    -   id: check-yaml
    -   id: check-added-large-files
```

### Error Logging Function
```python
def log_aws_initialization_error(error):
    logger = create_logger(__name__)
    logger.critical(f"AWS S3 Initialization Failed: {error}")
    logger.critical("Troubleshooting:")
    logger.critical("1. Verify AWS credentials")
    logger.critical("2. Check IAM user permissions")
```

### AWS Credentials Validation
```python
try:
    validate_config()
    validate_aws_credentials()
except ConfigurationError:
    sys.exit(1)
```

## Testing
- Verified error logging
- Confirmed pre-commit hooks
- Maintained error context

### CODE REVIEW

**Review by pedroheyerdahl — CHANGES_REQUESTED** [review]
I've reviewed the changes and have some suggestions:

Logging Consolidation:
We currently have duplicate logger creation methods (setup_logger() in utils.py and create_logger() in logging_config.py)
I recommend removing setup_logger() and enhancing create_logger() to include file logging support
This would make logging_config.py our single source of truth for logging configuration
Several files still reference setup_logger(), including upload/run.py - these would need updating
Exception Handling:
The custom exceptions provide good error categorization, but their implementation is inconsistent across files
Rather than having exceptions defined in multiple files, we should consolidate them in exceptions.py
Files like ingest/run.py should import and use these centralized exceptions

File: setup.cfg
[the remark this answers is not in the document]
Comment (db-tycoon-stephen) [reply setup.cfg:1]: @pedroheyerdahl do you believe that ruff is sufficient? If so I'll take out the pre-commit files

File: src/pipeline/config.py [hunk 1]
```diff
@@ -9,7 +15,7 @@
     # RAW_DATA_DIR = os.path.join(ROOT_DIR, 'raw_data')
     # PROC_DATA_DIR = os.path.join(ROOT_DIR, 'processed')
 
-DATALAKE_DIR = os.path.join(ROOT_DIR, 'datalake')
+DATALAKE_DIR = os.path.join(ROOT_DIR, 'data')
```
Comment (db-tycoon-stephen) [inline src/pipeline/config.py:18]:
```suggestion
DATALAKE_DIR = os.path.join(ROOT_DIR, 'datalake')
```

File: src/pipeline/catalog.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import ibis
 from pipeline.utils import setup_logger
```
[the remark this answers is not in the document]
Comment (db-tycoon-stephen) [reply src/pipeline/catalog.py]: Removed

**Comment by db-tycoon-stephen** [comment]
I will go through and resolve all the merge conflicts once we all feel good about the new code we're adding

**Comment by db-tycoon-stephen** [comment]
Merging after a slack approval from Pedro

**Review by StuffbyYuki — APPROVED** [review]

**Pull request merged**